app/models/cloud_volume.rb

```diff
@@ -23,10 +23,20 @@ def self.available
     left_outer_joins(:attachments).where("disks.backing_id" => nil)
   end
 
-  def self.class_by_ems(ext_management_system)
+  def self.class_by_ems(ems)
     # TODO(lsmola) taken from OrchesTration stacks, correct approach should be to have a factory on ExtManagementSystem
     # side, that would return correct class for each provider
-    ext_management_system && ext_management_system.class::CloudVolume
+
+    # If no EMS was given, fallback to nil
+    return nil unless ems
+
+    klass = ems.class::CloudVolume
+    # Return it if this is the actuall class (different from base CloudVolume)
+    return klass unless klass.name == 'CloudVolume'
```

Comment:
A simpler way would be to have just `return klass unless klass == CloudVolume`.

A bit more verbose version would be:

```
    klass_name = "#{ems.class}::CloudVolume"
    klass = Module.const_get(klass_name)
    # Return it if this is the actuall class (different from base CloudVolume)
    return klass if klass.name == klass_name
```

This first stores the name of the target class, then looks for the class and in the last row checks whether the name of the retrieved class is the target class. The problem is that `Module.const_get` (or just `ems.class::CloudVolume`) return base `CloudVolume` in case the target class does not exist.

All provided tests pass with any of these alternatives.

I was looking for other means to check if the class actually exists, but none of them worked here. If there are other ways to get around this, I'd be happy to understand them.